tests/datasets/test_dataset.py

```diff
@@ -0,0 +1,54 @@
+from abc import abstractmethod
+import pytest
+
+from czbenchmarks.datasets.dataset import Dataset
+
+class DatasetTests:
+    """Tests for Dataset class that should be tested for each concrete subclass. This is accomplished by having the concrete subclass extend this class and implement the `valid_dataset` fixture, which returns a valid instance of the dataset class, used by the tests herein."""
```

Comment:
This is an interesting way of doing it. I've never seen it before. I probably would have done it with something like `pytest.fixture(params=[SingleCellLabeledDataset, ...])`. Does this have any advantages over that?

Edit: it seems like the main advantage is you can also define other helper methods to actually put data in place for the fixture within the subclasses?